activesupport/lib/active_support/correctable_error.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module ActiveSupport #:nodoc
+  module CorrectableError #:nodoc
+    extend Concern
+
+    module ClassMethods #:nodoc
+      def correct_error(correction_class) # :nodoc:
+        # We may not have DidYouMean
+        if defined?(DidYouMean)
```

Comment:
Do we still need this condition? ` && DidYouMean.respond_to?(:correct_error)`

Reply:
It's public api since DidYouMean 1.4.0. This is included by Ruby 2.7.
So I think we don't need to check it.